core/unit_test/TestTeamVector.hpp

```diff
@@ -1016,7 +1016,7 @@ struct checkScan {
 };
 }  // namespace VectorScanReducer
 
-#if !(defined(KOKKOS_IMPL_CUDA_CLANG_WORKAROUND) || defined(KOKKOS_ENABLE_HIP))
+#if !defined(KOKKOS_IMPL_CUDA_CLANG_WORKAROUND)
```

Comment:
If it works for HIP and CUDA nvcc, there is a high probability that it also works for CUDA clang. Not for this PR but I think it's worth to check if the still need this condition.